### PR TITLE
Refactor yaml

### PR DESCRIPTION
**Pull request by jessicavers**
Yaml generator modified to allow more recent yaml structure. 
Yaml templates updated.

### CODE REVIEW

File: templates/yaml_templates_generator.py [hunk 1]
```diff
@@ -115,25 +85,111 @@ def yaml_generator(path_to_modules: str, output_folder: str) -> int:
...
+        v: Method value, dict value
+        params_dict: Parameter dictionary
+    """
+    if str(v).find("=") == -1 and str(k) != "kwargs":
+        params_dict[str(k)] = "REQUIRED"
+    elif str(k) == "kwargs":
+        params_dict["#additional parameters"] = "AVAILABLE"
+    elif str(k) == "center":
+        # Temporary value
+        params_dict[str(k)] = "$ {{centering.side_outputs.centre_of_rotation}}"
```
Comment (yousefmoazzam) [inline templates/yaml_templates_generator.py:113]: Tiny nitpick: github action YAML syntax has no space between the dollar sign and the curly braces when making references to other values (similar to substitution in sh/bash) https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-using-a-single-dimension-matrix, so I think this should maybe be changed from
```
"$ {{centering.side_outputs.centre_of_rotation}}"
```

to
```
"${{centering.side_outputs.centre_of_rotation}}"
```

That change will probably make the parsing of the string a bit easier (ie, detecting the `${{}}`) when it comes to it, leaving the space in there could make things a bit more difficult that it needs to be?

Comment (jessicavers) [reply templates/yaml_templates_generator.py]: Thank you for reading through and picking this up

Comment (yousefmoazzam) [reply templates/yaml_templates_generator.py]: No problem, was happy to read through it. The YAML template generation logic separated into a few more functions looks good :slightly_smiling_face:

**Comment by dkazanc** [comment]
another little thing. Can we make the order for `side_outputs` to be the following:
```
side_outputs:
    side_output_name: reference_name
```
As it is presented [here](https://github.com/DiamondLightSource/httomo/blob/d9d78d4ae86b17749fcc251a54e0e4fc7ac2e806/samples/pipeline_template_examples/new_syntax_pipeline.yaml#L31). This makes it consistent with the changes that the contractor did. Thanks

File: templates/httomo/httomo.data.hdf.loaders/standard_tomo.yaml [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
-      dimension: 1
-      preview: [None, None, None]
-      pad: 0
+- method: standard_tomo
+  module_path: httomo.data.hdf.loaders
+  parameters:
+    name: tomo
+    data_path: entry1/tomo_entry/data/data
+    image_key_path: entry1/tomo_entry/instrument/detector/image_key
+    angles_path: /entry1/tomo_entry/data/rotation_angle
```
[the remark this answers is not in the document]
Comment (jessicavers) [reply templates/httomo/httomo.data.hdf.loaders/standard_tomo.yaml:7]: Yes, updated. Thank you for noticing that

**Comment by jessicavers** [comment]
To confirm, the yaml template for a parameter containing side outputs would be:

```
- method: find_center_vo
  side_outputs:
    cor : centre_of_rotation
- method: recon
  parameters:
    center: ${{centering.side_outputs.centre_of_rotation}}
```
instead of referencing the dict keys only:
```
- method: find_center_vo
  side_outputs:
    centre_of_rotation : cor
- method: recon
  parameters:
    center: ${{centering.side_outputs.centre_of_rotation}}
```

File: templates/httomo/httomo.data.hdf.loaders/standard_tomo.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jessicavers) [reply templates/httomo/httomo.data.hdf.loaders/standard_tomo.yaml:7]:
```
- method: standard_tomo
  module_path: httomo.data.hdf.loaders
  parameters:
    name: REQUIRED
    in_file: REQUIRED
    data_path: REQUIRED
    dimension: REQUIRED
    preview: REQUIRED
    pad: REQUIRED
    comm: REQUIRED
    image_key_path: null
    rotation_angles:
      data_path: /entry1/tomo_entry/data/rotation_angle
    ignore_darks: false
    ignore_flats: false
```
To ask an obvious question, should each sample yaml template include an in_file value such as tests/test_data/tomo_standard.nxs now, following the standard _tomo updates?

[the remark this answers is not in the document]
Comment (dkazanc) [reply templates/httomo/httomo.data.hdf.loaders/standard_tomo.yaml]: yep, I'd go with our conventional parameters as for `tomo_standard` dataset. By the way, `comm` is a private argument I think, so it should be removed from the template. Actually anywhere you see it in functions, the yaml generator should ignore it, sort of like `ncore` or `ncores`. 
```
- method: standard_tomo
  module_path: httomo.data.hdf.loaders
  parameters: 
    name: tomo
    data_path: entry1/tomo_entry/data/data
    image_key_path: entry1/tomo_entry/instrument/detector/image_key
    rotation_angles: {"data_path": "/entry1/tomo_entry/data/rotation_angle"}
    dimension: 1
    preview:
      - 
      - 
      - 
    pad: 0
```

[the remark this answers is not in the document]
Comment (dkazanc) [reply templates/httomo/httomo.data.hdf.loaders/standard_tomo.yaml]: `rotation_angles: {"data_path": "/entry1/tomo_entry/data/rotation_angle"} ` is equivalent to 
```
rotation_angles:
    data_path: /entry1/tomo_entry/data/rotation_angle
```
I'd go for the latter as you have...

[the remark this answers is not in the document]
Comment (jessicavers) [reply templates/httomo/httomo.data.hdf.loaders/standard_tomo.yaml]: Great, sounds good, thank you

**Comment by dkazanc** [comment]
@jessicavers thanks, I'm thinking to merge it to [refactor-yaml](https://github.com/DiamondLightSource/httomo/tree/refactor-yaml) and will work on that branch to see that tests run etc. is that OK?

**Comment by jessicavers** [comment]
If there is time I will just add the in_file parameter to the sample yaml templates

**Pull request merged**